reposerver/src/main/scala/com/advancedtelematic/tuf/reposerver/target_store/S3TargetStoreEngine.scala

```diff
@@ -29,10 +30,20 @@ class S3TargetStoreEngine(credentials: S3Credentials)(implicit val system: Actor
 
   private val log = LoggerFactory.getLogger(this.getClass)
 
-  private lazy val s3client = AmazonS3ClientBuilder.standard()
-    .withCredentials(credentials)
-    .withRegion(credentials.region)
-    .build()
+  protected lazy val s3client = {
+    if(credentials.endpointUrl.length() > 0) {
```

Comment:
this could be:

```scala
  protected lazy val s3client = {
    val builder = AmazonS3ClientBuilder.standard().withCredentials(credentials)
    
    if(credentials.endpointUrl.nonEmpty)
      builder.withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(credentials.endpointUrl, credentials.region.getName))
    else
      builder
  }.build()
```

Reply:
```else 
  builder.withRegion(credentials.region)
```
?